db/similarity.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import absolute_import
+
+import db
+from db.data import count_all_lowlevel
+from db.exceptions import NoDataFoundException, BadDataException
+import similarity.metrics
+import similarity.utils
+
+from sqlalchemy import text
+
+
+def add_metrics(batch_size):
+    """Computes each metric from the similarity_metrics table for
+    each recording in the lowlevel table, in batches.
+    """
+    lowlevel_count = count_all_lowlevel()
+
+    with db.engine.connect() as connection:
+        metrics = similarity.utils.init_metrics()
+        sim_count = count_similarity()
+        print("Processed {} / {} ({:.3f}%)".format(sim_count,
+                                                   lowlevel_count,
+                                                   float(sim_count) / lowlevel_count * 100))
+        
+        batch_query = text("""
```

Comment:
@alastair we need to get both the low and high level data. I altered this query to collect it all in bulk. I think it has made our querying much faster - it takes ~28.5 seconds to do a 10 000 recording batch, so near 10h for the whole 12m recording database. I think this is an alternative approach to the query as well:

```
            SELECT id, llj.data AS ll_data, jsonb_object_agg(hl.model, hl.data) AS hl_data
              FROM lowlevel
         LEFT JOIN lowlevel_json AS llj
             USING (id)
         LEFT JOIN highlevel_model AS hl
             USING (id)
         LEFT JOIN similarity
             USING (id)
             WHERE similarity.id IS NULL GROUP BY (id, llj.data)
             LIMIT :batch_size
```
I used explain and tested both, the one that I implemented seems slightly less complex and took about 2 seconds less per batch.

A further improvement could be using some code from #347 to select only some features rather than the whole lowlevel document, which we don't need.